app/pb_kits/playbook/pb_checkbox/checkbox.rb

```diff
@@ -2,93 +2,42 @@
 
 module Playbook
   module PbCheckbox
-    class Checkbox < Playbook::PbKit::Base
-      PROPS = %i[configured_classname
-                 configured_data
-                 configured_id
-                 configured_dark
-                 configured_text
-                configured_value
-              configured_name
-              configured_checked
-              configured_icon
-            ].freeze
+    class Checkbox
+      include Playbook::Props
 
-      def initialize(classname: default_configuration,
-                     data: default_configuration,
-                     id: default_configuration,
-                     dark: default_configuration,
-                     text: default_configuration,
-                   value: default_configuration,
-                 name: default_configuration,
-                 checked: default_configuration,
-                 icon: default_configuration
-               )
-        self.configured_classname = classname
-        self.configured_data = data
-        self.configured_id = id
-        self.configured_dark = dark
-        self.configured_text = text
-        self.configured_value = value
-        self.configured_name = name
-        self.configured_checked = checked
-        self.configured_icon = icon
+      partial "pb_checkbox/checkbox"
 
-      end
-
-      def to_partial_path
-        "pb_checkbox/checkbox"
-      end
-
-      def icon
-        if is_set? configured_icon
-          pb_icon = Playbook::PbIcon::Icon.new(icon: "check", id: "check_icon", classname: "check_icon", fixed_width: true)
-          ApplicationController.renderer.render(partial: pb_icon, as: :object)
-        end
-      end
-
-      def text
-        default_value(configured_text, "")
-      end
+      prop :dark, type: Playbook::Props::Boolean, default: false
+      prop :checked, type: Playbook::Props::Boolean, default: false
+      prop :icon, type: Playbook::Props::Boolean, default: false
+      prop :text, default: ""
+      prop :value, default: ""
+      prop :name, default: ""
```

Comment:
Are these `default` needed?

Reply:
Looking at the template, I think no. I will remove them.